src/const_choice.rs

```diff
@@ -71,10 +71,19 @@ impl ConstChoice {
     /// Returns the truthy value if `x < y`, and the falsy value otherwise.
     #[inline]
     pub(crate) const fn from_word_lt(x: Word, y: Word) -> Self {
+        // See "Hacker's Delight" 2nd ed, section 2-12 (Comparison predicates)
         let bit = (((!x) & y) | (((!x) | y) & (x.wrapping_sub(y)))) >> (Word::BITS - 1);
```

Comment:
This is also duplicated here, FWIW: https://github.com/RustCrypto/crypto-bigint/blob/c529c86/src/modular/boxed_residue/mul.rs#L251

Reply:
It is based on the same algorithm, but it is not exactly the same - it cannot make use of `ConstChoice::from_word_lt()`

Reply:
Maybe we could provide these two algorithms implemented in a more general form in a common module somewhere? (a new private toplevel `bits` module, perhaps?)

Reply:
I thought that it would be nice to have some kind of a "primitives" module with all the little algorithms for bit magic, operations on `Word`s with carry, and so one, which are currently spread around the crate. So that people knew where to find them and didn't write their own versions.

Reply:
Sounds good to me

Reply:
I suppose that should be a separate PR?

Reply:
yeah that would be helpful with my multiplication PR as well

Reply:
@fjarri whatever works, just a suggestion

Reply:
Working on the primitives module in #458